network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/accordion_block.html

```diff
@@ -5,7 +5,7 @@
 
     <div class="tw-row">
         {% for block in self.accordion_items %}
-            <details class="tw-w-full tw-font-sans tw-font-light tw-mx-8">
+            <details class="article-content:tw-w-4/5 article-content:tw-mx-auto tw-w-full tw-font-sans tw-font-light tw-mx-8">
```

Comment:
What do you think about moving these things out into the article page? I think it usually makes it harder to understand and maintain components if they need to know where they are being used. Rather, components should be like pure functions that don't have any side effects or know anything about their surrounding. 

In this case it seems like on article page you want to limit the block to a given width and center it. What do you thin about doing this via a wrapper on the article page? We would not need any of the Tailwind changes.

Reply:
But the accordions are streamfields. If the accordions were manually added to article pages in a predictable spot on the article pages, I would agree about adding a wrapper, but they will be mixed up with other CMS content. Still, I would be adding to either `network-api/networkapi/templates/pages/buyersguide/article_page.html#L49` and adding an if statement about accordions on there but you need to apply the width changes after the streamfield-content class since that class has different flex/width.  

Or does that mean we are modifying `network-api\networkapi\wagtailpages\templates\wagtailpages\blocks\base_streamfield_block.html`  and adding checks to accordions there? Should I just make the CSS changes on `source/sass/wagtail/blocks/article-blocks.scss` ? The whole purpose of adding these variants is to make stylings from different parent element changes easier to follow.  I definitely need more specific instructions moving forward with this idea.

Reply:
The accordions should be as wide as the rest of the blocks on the article page. Why are they wider now? Why are other blocks not wider? The width limitation should come from the article page and not from the blocks to make the blocks reusable. So yea, the change should probably be in the article page template. 

Components in general should not really know about their surrounding. Rather everything that is variable about a component should be passed in directly as flags. It makes components otherwise really hard to maintain because they include all these checks regarding where they are used. It's better to make these flags/checks semantic about the effect (in this case it should express something about the width) instead of the environment in which the flag should be set. A good analogy for components are pure functions, same input should lead to same output. Also, there should be no side effects should happen and they should not "pull" data, all data should be passed in as arguments. 

Also, I think that "variants" is the bit misleading place for this one off use case. Variants are widely available and should have something to do with styling. "Article page" is not really a concept in CSS. If we wanted to do this width adjustment in the component, which I still think is not the right way to do it, there should just be a flag that is passed into the component template which then applies that class or not. The flag should also not be `article_page=True/False` but rather `width='4_5'` or something of that nature. Or you even allow passing of extra classes to the component. 

If you need to handle a particluar block in a streamfield differently based on it's type, you can check the blocks type with `block.block_type`. https://docs.wagtail.org/en/stable/topics/streamfield.html#template-rendering

Reply:
Adding a wrapper on the article page does not work because `base_streamfield_block.html` has container classes so the parent having 4/5 width would not achieve the same effect.

redid the solution by applying a width='4_5' value.